### ISSUE TITLE
Add modal to display error messages

### ISSUE THREAD

**Issue by jleach** [bcgov/bc-wallet-mobile#151]
As a user, I would like to see a modal overlay when an error occurs so I better understand my current situation.

Wireframes for in-app errors
https://xd.adobe.com/view/4f9b250e-27d3-4d8b-af48-c6b90520d0ee-1ac8/ 

Wireframes for errors during scanning
https://xd.adobe.com/view/cfc523f6-fb60-45fd-bb3d-a09fa3d04c98-9a96/

### Acceptance Criteria
Given the Holder within the BC Wallet app
When BC Wallet malfunctions due to system errors
Then the Holder is presented with an error modal overlay describing the error event in layman terms,
And an error code
And an "Okay" button that dismisses the error modal overlay

**Comment by knguyenBC** [comment]
@jleach I'm unsure as to how I might be able to test this as I'm not sure how to force malfunctions.
